pkg/monitoring/vmstats/collector.go

```diff
@@ -146,34 +167,101 @@ type prometheusScraper struct {
 	ch chan<- prometheus.Metric
 }
 
-func (ps *prometheusScraper) Report(vms []*k6tv1.VirtualMachine) {
+func (ps *prometheusScraper) Report(co *VMCollector, vms []*k6tv1.VirtualMachine) {
 	for _, vm := range vms {
-		ps.updateVMMetrics(vm)
+		ps.updateVMStatusMetrics(vm)
+		ps.updatePVInfoMetrics(co, vm)
+	}
+}
+
+func (ps *prometheusScraper) updatePVInfoMetrics(co *VMCollector, vm *k6tv1.VirtualMachine) {
+	if co.pvInformer == nil || co.pvcInformer == nil || vm.Spec.Template.Spec.Volumes == nil {
+		return
+	}
+
+	for _, volume := range vm.Spec.Template.Spec.Volumes {
+		if volume.PersistentVolumeClaim == nil {
+			continue
+		}
+		log.Log.V(4).Infof("VM %s/%s has PVC: '%s'", vm.Namespace, vm.Name, volume.PersistentVolumeClaim.ClaimName)
+
+		pvc, err := getPVC(co, vm, volume.PersistentVolumeClaim.ClaimName)
+		if err != nil {
+			continue
+		}
+		pv, err := getPV(co, vm, pvc)
+		if err != nil {
+			continue
+		}
+
+		ps.pushMetric(metrics[pvInfoMetric], prometheus.GaugeValue, 1,
```

Comment:
Following the [comment](https://github.com/kubevirt/kubevirt/pull/9741#issuecomment-1612535963) you need to check whether the PV type is block, you will have to pass this information as well.

Reply:
done